packages/renderer/src/lib/ui/Typeahead.svelte

```diff
@@ -40,26 +42,66 @@ let input: HTMLInputElement | undefined = $state();
 let list: HTMLDivElement | undefined = $state();
 let scrollElements: HTMLElement[] = $state([]);
 let value: string = $state('');
-let items: string[] = $derived(
-  resultItems.toSorted(
-    compare ??
-      ((a: string, b: string): number => {
-        if (a.startsWith(userValue) === b.startsWith(userValue)) {
-          return a.localeCompare(b);
-        } else if (a.startsWith(userValue) && !b.startsWith(userValue)) {
-          return -1;
-        } else {
-          return 1;
-        }
-      }),
-  ),
-);
 let opened: boolean = $state(false);
 let highlightIndex: number = $state(-1);
 let pageStep: number = $state(10);
 let userValue: string = $state('');
 let loading: boolean = $state(false);
 
+let groupedItems: { [group: string]: string[] } = $derived(groupItems(resultItems));
+let itemHeadings: { [index: number]: string[] } = $derived(updateHeadings(groupedItems));
```

Comment:
as these two derived state are the only callers of the function, I would inline these function in the derived block

$derived.by(() => inline the function 

idem for updateHeadings